File: edutask-core/src/main/java/ru/usernamedrew/edutaskcore/entity/Task.java
package ru.usernamedrew.edutaskcore.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name = "task")
public class Task extends BaseEntity {
    @Column(nullable = false)
    private String title;

    @Column(nullable = false, columnDefinition = "text")
    private String statement;

    @Column(name = "input_format", columnDefinition = "text")
    private String inputFormat;

    @Column(name = "output_format", columnDefinition = "text")
    private String outputFormat;

    @Enumerated(EnumType.STRING)
    @Column(nullable = false, length = 10)
    private TaskDifficulty difficulty;

    @ManyToOne(fetch = FetchType.LAZY, optional = false)
    @JoinColumn(name = "author_id", nullable = false)
    private UserProfile author;

    @ManyToMany
    @JoinTable(
        name = "task_topic",
        joinColumns = @JoinColumn(name = "task_id"),
        inverseJoinColumns = @JoinColumn(name = "topic_id")
    )
    private Set<Topic> topics = new HashSet<>();

    @ManyToMany
    @JoinTable(
        name = "task_language",
        joinColumns = @JoinColumn(name = "task_id"),
        inverseJoinColumns = @JoinColumn(name = "language_id")
    )
    private Set<ProgrammingLanguage> supportedLanguages = new HashSet<>();

    public enum TaskDifficulty {
        EASY, MEDIUM, HARD
    }
}

File: edutask-frontend/src/types/task.ts
export type TaskDifficulty = 'EASY' | 'MEDIUM' | 'HARD';

export type TaskSummary = {
  id: string;
  title: string;
  difficulty: TaskDifficulty;
};

export type TaskTopicSummary = {
  id: string;
  name: string;
};

export type TaskResponse = {
  id: string;
  title: string;
  statement: string;
  inputFormat: string | null;
  outputFormat: string | null;
  difficulty: TaskDifficulty;
  authorId: string;
  topics: TaskTopicSummary[];
  createdAt: string;
  updatedAt: string;
};

export type TaskCreateRequest = {
  title: string;
  statement: string;
  inputFormat?: string;
  outputFormat?: string;
  difficulty: TaskDifficulty;
  topicIds: string[];
};

export type TaskUpdateRequest = Partial<TaskCreateRequest>;
